docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,789 @@
+- For each of the signers, a Bitcoin address to receive the split amount.
+
+==== Required Inputs
+
+From this, we reach the full set of required inputs and where they can be
+procured or computed:
+
+- N, the number of signers in the group, available on the Keep contract
+  from the `getMembers` view method, but also tracked locally on the client
+  alongside the members of the keep.
+- L, the deposit lot size, available on the tBTC deposit contract from the
+  `lotSizeSatoshis` view method.
+- The previous outpoint describing the deposit UTXO, available on the tBTC
+  deposit contract from the `fundingInfo` view method, which returns it as
+  the third `utxoOutpoint` entry in its 3-part return tuple.
+- The keep's public key, available on the Keep contract from the
+  `getPublicKey` view method.
+- The deposit redemption script, straightforwardly computable from the keep's
+  public key by compressing the public key; this is briefly covered in the
+  implementation <<output-script>> section.
+- One Bitcoin address per signer, specified on the client in a way detailed
+  in the section on <<recovery-address>>.
+- The keep's signature on the sighash produced by the previous items,
+  computed via the threshold ECDSA protocol in a way detailed in the section on
+  the <<signing-protocol>>.
+- F, the desired fee, which is a function of the going feerate on the Bitcoin
+  network and the number of bytes in the final transaction.
+
+The desired fee is the most problematic one to procure under the goal of not
+strictly requiring a Bitcoin node connection, as generally a Bitcoin node
+provides estimates of the feerates that will achieve transaction confirmation
+within a certain amount of time. A short-term solution is proposed in the
+section on <<bitcoin-fee>>.
+
+[[recovery-address]]
+===== Specifying an Operator's Bitcoin Recovery Address
+
+To specify a given operator's Bitcoin Recovery Address, this proposal
+suggests adding a single key to the `Extensions.TBTC.BTCRefunds` config
+entry, `BeneficiaryAddress`. The value for this key should be one of:
+
+- A simple BTC address.
+- An xpub, ypub, or zpub that allows the client to derive new addresses for
+  each refund.
+
+The challenge with a *pub is that, to avoid reuse, it needs to do one of two
+things:
+
+- Be able to check, on-chain, whether a given address has been used; or,
+- Have a *pub that is only meant for use by the client, and have the client
+  track the latest address generated (as an index starting at 0).
+
+To avoid _strictly requiring_ a connection to a Bitcoin node, this RFC
+proposes tracking the latest address generated, and allowing an additional
+config key, `TargetDerivationStartIndex`, that sets a starting point for such
+derivations. The client should write the updated index to the `current/`
+directory that tracks active key shares, to an encrypted file named
+`current-btc-refund-derivation-index`. The operator should then seek to
```

Comment:
If the storage contains updated index, is `TargetDerivationStartIndex` in config ignored? Can we run into a situation when the value held by the storage is not up to date? For example, the situation when the receiving party transferred BTC they obtained somewhere else.

Reply:
I'll clarify that the way to disambiguate between storage value and config value is to use the higher of the two.

I'm not sure I follow the scenario you're describing for the second question. The value held by the storage could be ”out of date“ in the sense of pointing to an already-used derived address if the recipient is using the same extended pubkey chain for more than just this purpose, or if there's a concurrency issue (feels solvable by an atomically incrementing int). Otherwise I don't see a clear way for that to happen. The optional Bitcoin node connection can help for the former scenario (subject to concurrency there as well, ofc).

Reply:
> if the recipient is using the same extended pubkey chain for more than just this purpose

This is the scenario that I meant.

Reply:
Yeah, there's only so much we can do there. In my mind it would make the most sense to recommend usage of an extended pubkey chain that is not used elsewhere, or to provide a dedicated branch of the derivation tree (e.g. `/0` is generally for external transactions and `/1` for internal ones, but you could provide `/2` if you wanted a dedicated chain). The function in the derivation implementation details does allow for this:

```
// In cases where the extended public key is at depth 4, meaning the external or
// internal chain is already included, DeriveAddress will directly derive the
// address index at the existing depth.
```

Reply:
Cool. Just to clarify: if the operator provides a simple BTC address as `BeneficiaryAddress`, receive there BTC from a liquidated deposit, and transfer this BTC out of this account to some other account. Do they have to change `BeneficiaryAddress` in the config for further recovery of BTC from a liquidated deposit? My understanding is that the answer is no, they do not have to change anything. I am not sure about *pub though.

Reply:
No, all of this derivation behavior only applies to extended pubkeys.

For extended pubkeys, we optionally (if there is a connection) check that there are no other transactions to the derived address, and derive a new one if there are existing transactions. The transaction count should be nonzero even if the address balance has been transferred out (that's why we check transaction count rather than balance or unspent outputs).

Reply:
Makes sense, thanks for the explanation!